src/clojure/contrib/humanize.cljc

```diff
@@ -297,3 +297,75 @@
 
      ;; FIXME:
      :else (to-string diff))))
+
+(def ^:private duration-periods
+  [[(* 1000 60 60 24 365) "year"]
+   [(* 1000 60 60 24 31) "month"]
+   [(* 1000 60 60 24 7) "week"]
+   [(* 1000 60 60 24) "day"]
+   [(* 1000 60 60) "hour"]
+   [(* 1000 60) "minute"]
+   [1000 "second"]])
+
+(defn duration-terms
```

Comment:
Originally, I wanted it public to allow for customizations on how the data was presented, but I ended up adding enough options to `duration` that, yes, it can be private.